views/clockviews/shared.html

```diff
@@ -30,19 +30,41 @@
     <div class="inner">
       <div id="trend">
         <div id="bgnow"></div>
-        <div id="arrowDiv"><img id="arrow" src=""/></div>
+        <div id="arrowDiv"><img id="arrow" src="" alt="arrow"/></div>
       </div>
       <div id="clock"></div>
       <div id="staleTime"></div>
     </div>
   </main>
-  
-  <script src="/api/v1/status.js"></script>
-  <script src="<%= locals.bundle %>/js/bundle.clock.js?v=<%= locals.cachebuster %>"></script>
-  
+  <script src="/<%= locals.bundle %>/js/bundle.clock.js?v=<%= locals.cachebuster %>"></script>
+
   <script type="text/javascript">
-    let clockFace = "<%= face  %>"; // can now be used in scripts
-    window.Nightscout.client.init( );
+    var clockFace = "<%= face  %>"; // can now be used in scripts
+
+    var parts = (location.search || '?').substring(1).split('&');
+    var token = '';
+    parts.forEach(function (val) {
+      if (val.startsWith('token=')) {
+        token = val.substring('token='.length);
+      }
+    });
+
+    var secret = typeof localStorage !== 'undefined' ? localStorage.getItem('apisecrethash') : '';
+    var src = '/api/v1/status.js?t=' + Date.now();
+
+    if (secret) {
+      src += '&secret=' + secret;
+    } else if (token) {
+      src += '&token=' + token;
+    }
+
+    var script = document.createElement('script');
+    script.onload = function () {
+      window.Nightscout.client.init( );
+    };
+    script.src = src;
+
+    document.head.appendChild(script); //or something of the likes
```

Comment:
Load JSON with token or secret for the case that authentication is enabled.


There is still an open case if the user starts directly to work with clock view without calling the index.html. In this case no authentication dialog will be shown. But I don't know such a use case.  In this case the user can use token.